spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java

```diff
@@ -135,4 +137,13 @@
 	 * @return the app deployment status
 	 */
 	AppStatus status(String id);
+
+	/**
+	 * Return the environment info for this deployer. This info should include the
+	 * following: 'deployer-class-name','deployer-version', 'deployer-spi-version', 'platform-type',
+	 * 'platform-client-version', 'platform-host-version', 'java-version', 'spring-version', 'spring-boot-version'.
+	 *
+	 * @return the app deployment environment info
+	 */
+	Map<String, String>  environmentInfo();
```

Comment:
OK, let me add a commit with a new return type, then we can argue what should be included.

I'll start with:

- deployerSpiVersion
- deployerImplementationVersion
- platformType
- platformClientVersion
- platformHostVersion
- javaVersion
- springVersion
- springBootVersion